From a58a023e9e207e30fbedc5f331be15306803b262 Mon Sep 17 00:00:00 2001
From: Nagaraj S <sivasankaranx.nagaraj@intel.com>
Date: Wed, 11 May 2016 17:12:57 -0700
Subject: [PATCH 396/441] usb/phy/phy-intel-cht.c: Disable extcon notifier

Register extcon notifier is not required for the OTG device.
extcon notifier is required only OTG dual role or host only mode.

Change-Id: I996d778dd42ffafecadbb83892da087a200dedd7
Solution: Disable extcon event registration for device only mode.
---
 drivers/usb/phy/phy-intel-cht.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/phy/phy-intel-cht.c
+++ b/drivers/usb/phy/phy-intel-cht.c
@@ -543,6 +543,7 @@ static int cht_otg_probe(struct platform
 	cht_otg_dev->nb.notifier_call = cht_otg_handle_notification;
 	usb_register_notifier(&cht_otg_dev->phy, &cht_otg_dev->nb);
 
+#if IS_ENABLED(CONFIG_USB_DWC3_HOST) || IS_ENABLED(CONFIG_USB_DWC3_DUAL_ROLE)
 	/* Register on extcon for OTG event too */
 	cht_otg_dev->id_nb.notifier_call = cht_handle_extcon_otg_event;
 	status = extcon_register_interest(&cht_otg_dev->cable_nb, NULL,
@@ -550,6 +551,7 @@ static int cht_otg_probe(struct platform
 	if (status)
 		dev_warn(&pdev->dev, "failed to register extcon notifier\n");
 
+#endif
 	/* init otg-fsm */
 	status = cht_otg_start(pdev);
 	if (status) {
@@ -583,7 +585,9 @@ err3:
 	device_remove_file(&pdev->dev, &dev_attr_cht_otg_state);
 err2:
 	cht_otg_stop(pdev);
+#if IS_ENABLED(CONFIG_USB_DWC3_HOST) || IS_ENABLED(CONFIG_USB_DWC3_DUAL_ROLE)
 	extcon_unregister_interest(&cht_otg_dev->cable_nb);
+#endif
 	usb_remove_phy(&cht_otg_dev->phy);
 err1:
 	kfree(cht_otg->phy.otg);
